diff --git a/targets/ipq806x-generic b/targets/ipq806x-generic
index 1b3263fc..eadf507f 100644
--- a/targets/ipq806x-generic
+++ b/targets/ipq806x-generic
@@ -26,3 +26,21 @@ device('netgear-nighthawk-x4s-r7800', 'netgear_r7800', {
 	factory_ext = '.img',
 	packages = QCA9984_PACKAGES,
 })
+
+device('netgear-nighthawk-xr500', 'netgear_xr500', {
+	factory_ext = '.img',
+	packages = QCA9984_PACKAGES,
+})
+
+
+-- Ubiquiti
+device('ubiquiti-unifi-ac-hd', 'ubnt_unifi-ac-hd', {
+	packages = QCA9984_PACKAGES,
+	factory = false,
+})
+
+
+-- Zyxel
+device('zyxel-nbg6817', 'zyxel_nbg6817', {
+	packages = QCA9984_PACKAGES,
+})
